### DISCUSSION TITLE
Closing popovers programmatically?

### DISCUSSION THREAD

**Discussion by viveleroi** [palantir/blueprint#6613] [Q&A] [answered]
We have a session timeout dialog that opens automatically after a period of inactivity, so there's no user interaction on the page.

If they've opened a blueprintjs popover, for example a Select component dropdown, it remains open when the dialog is open. I need a way to programmatically close all blueprintjs popovers.

I thought I could fake it by triggering `document.body.click()` but that doesn't seem to work. Are there any hooks or anything I can use to do this? I don't see anything noted in the docs.

**Comment by adidahiya** [comment, marked as answer]
> I thought I could fake it by triggering `document.body.click()` but that doesn't seem to work. Are there any hooks or anything I can use to do this? I don't see anything noted in the docs.

Popovers use this Overlay behavior to close on document "mousedown" events:

https://github.com/palantir/blueprint/blob/d8a5e6ae95884ae9866bf717a926411543341597/packages/core/src/components/overlay/overlay.tsx#L594-L596

You could try `document.dispatchEvent(new MouseEvent("mousedown"))` to close an open popover. This works for me on the [Popover docs page](https://blueprintjs.com/docs/#core/components/popover):


https://github.com/palantir/blueprint/assets/723999/72c41fa5-81fb-482c-b556-313ea838ffa5
